cEP-0022.md

```diff
@@ -0,0 +1,497 @@
+  to the project i.e. many optional bear settings are skipped and even many
+  bears are skipped due to non-availability of *"guessing"* features.
+- The errors pointed out by coala using generated config files
+  are generally undesirable for the project owners due to many organizations
+  following their own custom code styles rather than following pre-defined
+  standards implemented by most linters and bears.
+
+# Goals
+
+- Preparing *"green"* configuration files which *agree* to the project's code
+  style and adapt to their standards, with least user interaction, producing
+  absolutely zero errors in the code base.
+- The configuration file should be as specific as possible i.e. can point out
+  maximum errors with least amount of manual edits in the future, for future
+  inconsistencies in commits.
+- Storing metadata in the new types of files and classes proposed to enhance
+  the config file produced for future runs.
+- Maintaining a Github repo for storing metadata of combinations of bear
+  setting values already in use by existing organisations and using them as
+  initial checks before resorting to Brute Force.
+- Automating the task of **Adding `.coafile` and CI to enforce it**
+  ([example](https://github.com/processing/processing/pull/4781))
+  to the level of
+  generating config files so it is easier for
+  newcomers to get involved to create easily mergeable Pull Request in other
+  communities, which leads to easier adoption of coala.
+
+# Implementation
+
+This project will be implemented in five phases:
+
+- [Classification of settings and Metadata
+  Generation](#classification-of-settings-and-metadata-generation)
+- [The General Operations](#general-operations)
+- [Acting upon the newly available
+  metadata](#acting-upon-the-newly-available-metadata)
+- [The Brute Force way and further optimization on the Brute
+  Force](#the-brute-force-way-and-further-optimization-on-the-brute-force)
+- [Learning from Real World
+  Projects](#learning-from-real-world-projects)
+
+# Classification of settings and Metadata Generation
+
+We categorize the settings into 4 different types:
+
+1. The ones that attain a bool value.*(From here onwards referred
+   to as [Type 1 or Type bool](#type-1-or-type-bool-settings) settings)*
+2. The ones that can possibly attain an infinite number of values.*(From
+   here onwards referred to as [Type 2 or Type
+   infinite](#type-2-or-type-infinite-settings) settings)*
+3. The ones that can attain some fixed discrete set of values.*(From here onwards
+   referred to as [Type 3 or Type discrete](#type-3-or-type-discrete-settings)
+   settings)*
+4. The settings which require an additional config file to be parsed
+   by `coala-quickstart`.*(From here onwards referred to as [Type 4
+   or Type config](#type-4-or-type-config-settings) settings)*
+
+## Type **1** or Type *bool* settings
+
+### How will they be guessed?
+Since these settings can obtain a bool value, either True or False, the
```

Comment:
Many bools are tertinery, includes None.
Are they excluded from type 1?

Reply:
let me give you an example:
take for eg. the setting: `use_spaces: bool` from `SpaceConsistencyBear`
so the project can either use spaces and get a green setting, or use tabs and again get a different setting, but if its a mixture, the setting has to be dropped (more on this as a reply on your next comment) i.e. takes the value `None`